homeassistant/components/plugwise/sensor.py

```diff
@@ -0,0 +1,434 @@
+        ENERGY_KILO_WATT_HOUR,
+        DEVICE_CLASS_POWER,
+        "mdi:gauge",
+    ],
+    "gas_consumed_interval": [
+        "Current Consumed Gas",
+        VOLUME_CUBIC_METERS,
+        None,
+        "mdi:gas-cylinder",
+    ],
+    "gas_consumed_cumulative": [
+        "Cumulative Consumed Gas",
+        VOLUME_CUBIC_METERS,
+        None,
+        "mdi:gauge",
+    ],
+    "net_electricity_point": [
+        "Current net Power",
+        POWER_WATT,
+        DEVICE_CLASS_POWER,
+        "mdi:solar-power",
+    ],
+    "net_electricity_cumulative": [
+        "Cumulative net Power",
+        ENERGY_KILO_WATT_HOUR,
+        DEVICE_CLASS_POWER,
+        "mdi:gauge",
+    ],
+}
+
+MISC_SENSOR_MAP = {
+    "battery": ATTR_BATTERY_LEVEL,
+    "illuminance": ATTR_ILLUMINANCE,
+    "modulation_level": [
+        "Heater Modulation Level",
+        UNIT_PERCENTAGE,
+        "modulation",
+        "mdi:percent",
+    ],
+    "valve_position": ["Valve Position", UNIT_PERCENTAGE, None, "mdi:valve"],
+    "water_pressure": ATTR_PRESSURE,
+}
+
+INDICATE_ACTIVE_LOCAL_DEVICE = [
+    "cooling_state",
+    "flame_state",
+]
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Smile sensors from a config entry."""
+    api = hass.data[DOMAIN][config_entry.entry_id]["api"]
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]["coordinator"]
+
+    entities = []
+    all_devices = api.get_all_devices()
+    single_thermostat = api.single_master_thermostat()
+    for dev_id, device_properties in all_devices.items():
+        data = api.get_device_data(dev_id)
+        for sensor, sensor_type in {
+            **TEMP_SENSOR_MAP,
+            **ENERGY_SENSOR_MAP,
+            **MISC_SENSOR_MAP,
+        }.items():
+            if sensor in data:
+                if data[sensor] is None:
+                    continue
+
+                if "power" in device_properties["types"]:
+                    model = None
+
+                    if "plug" in device_properties["types"]:
+                        model = "Metered Switch"
+
+                    entities.append(
+                        PwPowerSensor(
+                            api,
+                            coordinator,
+                            device_properties["name"],
+                            dev_id,
+                            sensor,
+                            sensor_type,
+                            model,
+                        )
+                    )
+                else:
+                    entities.append(
+                        PwThermostatSensor(
+                            api,
+                            coordinator,
+                            device_properties["name"],
+                            dev_id,
+                            sensor,
+                            sensor_type,
+                        )
+                    )
+                _LOGGER.info("Added sensor %s", device_properties["name"])
```

Comment:
Please move the logging out of the loop. We can log the list of entities since the entity instance has a repr magic method.

Please lower to debug too.

Reply:
Will remove